### PR TITLE
SB-1119 performance improvements [in progress]

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java [hunk 1]
```diff
@@ -414,7 +414,7 @@ public EntityTransaction getTransaction()
     @Override
     public void close()
     {
-        database.close();
+        //database.close();
```
Comment (fuss86) [inline strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java]: `ODatabaseException: The database instance is not set in the current thread.`

File: strongbox-data-service/src/main/java/org/carlspring/strongbox/data/service/CommonCrudService.java [hunk 2]
```diff
@@ -49,7 +59,7 @@ public void postConstruct()
     {
         identifyEntity(entity);
         
-        ReflectionUtils.doWithFields(entity.getClass(), (field) -> {
+        ReflectionUtils.doWithFields(this.getEntityClass(), (field) -> {
```
Comment (sbespalov) [inline strongbox-data-service/src/main/java/org/carlspring/strongbox/data/service/CommonCrudService.java]: I think we can try to drop this `getEntityClass()` method and use some utility which will expose target class from underlying `InvocationHandler`.

**Comment by carlspring** [comment]
@fuss86,

Is this now obsolete?

**Comment by fuss86** [comment]
> Is this now obsolete?

Please leave it as it is for the time being :)

**Comment by fuss86** [comment]
Fixed in other PRs.

**Pull request closed without merging**